sdk/servicebus/service-bus/src/core/managementClient.ts

```diff
@@ -399,66 +496,73 @@ export class ManagementClient extends LinkEntity {
-          operation: Constants.operations.peekMessage
-        }
-      };
-      const associatedLinkName = this._getAssociatedReceiverName(this._context, sessionId);
-      if (associatedLinkName) {
-        request.application_properties![Constants.associatedLinkName] = associatedLinkName;
-      }
-      request.application_properties![Constants.trackingId] = generate_uuid();
-      log.mgmt(
-        "[%s] Peek by sequence number request body: %O.",
-        this._context.namespace.connectionId,
-        request.body
-      );
-      log.mgmt(
-        "[%s] Acquiring lock to get the management req res link.",
-        this._context.namespace.connectionId
-      );
-      await defaultLock.acquire(this.managementLock, () => {
-        return this._init();
-      });
+    const peekBySequenceNumberOperationPromise = () =>
+      new Promise<ReceivedMessage[]>(async (resolve, reject) => {
+        try {
+          const retryTimeoutInMs = getRetryAttemptTimeoutInMs(retryOptions);
+          const messageList: ReceivedMessage[] = [];
+          const messageBody: any = {};
+          messageBody[Constants.fromSequenceNumber] = types.wrap_long(
+            Buffer.from(fromSequenceNumber.toBytesBE())
+          );
+          messageBody[Constants.messageCount] = types.wrap_int(maxMessageCount!);
+          if (sessionId != undefined) {
+            messageBody[Constants.sessionIdMapKey] = sessionId;
+          }
+          const request: AmqpMessage = {
+            body: messageBody,
+            message_id: generate_uuid(),
+            reply_to: this.replyTo,
+            application_properties: {
+              operation: Constants.operations.peekMessage
+            }
+          };
+          const associatedLinkName = this._getAssociatedReceiverName(this._context, sessionId);
+          if (associatedLinkName) {
+            request.application_properties![Constants.associatedLinkName] = associatedLinkName;
+          }
+          request.application_properties![Constants.trackingId] = generate_uuid();
+          log.mgmt(
+            "[%s] Peek by sequence number request body: %O.",
+            this._context.namespace.connectionId,
+            request.body
+          );
 
-      const result = await this._mgmtReqResLink!.sendRequest(request);
-      if (result.application_properties!.statusCode !== 204) {
-        const messages = result.body.messages as { message: Buffer }[];
-        for (const msg of messages) {
-          const decodedMessage = RheaMessageUtil.decode(msg.message);
-          const message = fromAmqpMessage(decodedMessage as any);
-          message.body = this._context.namespace.dataTransformer.decode(message.body);
-          messageList.push(message);
-          this._lastPeekedSequenceNumber = message.sequenceNumber!;
+          const result = await this._acquireLockAndSendRequest(request, retryTimeoutInMs, {
+            abortSignal: undefined,
+            requestName: undefined
```

Comment:
Are these specified as `undefined` as a reminder to pipe these fields through in the future? Otherwise they seem unnecessary.

Reply:
Yes, another PR for `abortSignal`.